src/qz/ws/PrintSocketClient.java

```diff
@@ -461,6 +475,97 @@ private void processMessage(Session session, JSONObject json, SocketConnection c
                 break;
             }
 
+            case FILE_START_LISTENING: {
+                FileParams fileParams = new FileParams(params);
+                Path absPath = FileUtilities.getAbsolutePath(params, shownCertificate, true);
+                FileIO fileIO = new FileIO(session, params, fileParams.getPath(), absPath);
+
+                if (connection.getFileListener(absPath) == null && !fileIO.isWatching()) {
+                    connection.addFileListener(absPath, fileIO);
+
+                    FileUtilities.setupListener(fileIO);
+                    sendResult(session, UID, null);
+                } else {
+                    sendError(session, UID, "Already listening to path events");
+                }
+
+                break;
+            }
+            case FILE_STOP_LISTENING: {
+                if (params.isNull("path")) {
+                    connection.removeAllFileListeners();
+                    sendResult(session, UID, null);
+                } else {
+                    FileParams fileParams = new FileParams(params);
+                    Path absPath = FileUtilities.createAbsolutePath(fileParams, shownCertificate);
+                    FileIO fileIO = connection.getFileListener(absPath);
+
+                    if (fileIO != null) {
+                        fileIO.close();
+                        FileWatcher.deregisterWatch(fileIO);
+                        connection.removeFileListener(absPath);
+                        sendResult(session, UID, null);
+                    } else {
+                        sendError(session, UID, "Not already listening to path events");
+                    }
+                }
+
+                break;
+            }
+            case FILE_LIST: {
+                Path absPath = FileUtilities.getAbsolutePath(params, shownCertificate, true);
+
+                if (Files.exists(absPath)) {
+                    if (Files.isDirectory(absPath)) {
+                        ArrayList<String> files = new ArrayList<>();
+                        Files.list(absPath).forEach(file -> files.add(file.getFileName().toString()));
+                        sendResult(session, UID, new JSONArray(files));
+                    } else {
+                        sendError(session, UID, "Path is not a directory");
```

Comment:
The errors aren't logged to the console at all which leaves little debugging information.  We should have something rather verbose exposed to the log file when an error occurs, including the offending `absPath`.